comp/forwarder/defaultforwarder/default_forwarder.go

```diff
@@ -208,9 +211,10 @@ type DefaultForwarder struct {
 }
 
 // NewDefaultForwarder returns a new DefaultForwarder.
-func NewDefaultForwarder(options *Options) *DefaultForwarder {
+func NewDefaultForwarder(config config.Component, options *Options) *DefaultForwarder {
```

Comment:
Is the plan to remove `NewDefaultForwarder` in a future change in favor of injecting an implementation conforming to the component interface? If so - maybe It's worth adding `TODO`s around the methods that will go away soon.

Reply:
Yes. This is part of `Reduce the API surface of /comp/forwarder as a lot of functions and structs are exported.`. Tracked by ASC-480.
Comment added: https://github.com/DataDog/datadog-agent/pull/16563/commits/3e76b0acf1f6db041d5853a1837024e497716e10.